app/handlers/sessions_callback.rb

```diff
@@ -102,24 +102,24 @@ def handle
         run(TransferAuthentications, authentication, current_user)
         status = (current_user.is_temp ? :new_user : :returning_user)
       else
-        outcome = run(CreateUserFromOmniauth, @auth_data)
+        outcome = run(CreateUserFromOmniauthData, @data)
         new_user = outcome.outputs[:user]
         run(TransferAuthentications, authentication, new_user)
-        sign_in(new_user)
+        sign_in!(new_user)
         status = :new_user
       end
 
     end
 
     if this_authentication_is_new
-      run(TransferOmniauthInformation, @auth_data, current_user)
+      run(AddEmailFromOmniauthData, @data, current_user)
```

Comment:
Actually, `TransferOmniauthData` is misleading. `TransferAuthentications` transfers `Authentication` from one user to another, but `TransferOmniauthData` does not transfer an `OmniauthData` at all. Rather, it gets some info from `OmniauthData` and adds it as a new `ContactInformation` for the `User`.

But I'll still do it if you want.